### PR TITLE
[SMALLFIX] make permission checker extendable

### PR DESCRIPTION
**Pull request by bf8086**
This change extracts interface from the default permission checker to make it extendable.

### CODE REVIEW

**Comment by bf8086** [comment]
@gpang @calvinjia PTAL.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Default implementation of permission checker.
+ */
+// TODO(peis): Migrate this class to a set of static functions.
+@NotThreadSafe // TODO(jiri): make thread-safe (c.f. ALLUXIO-1664)
+public final class DefaultPermissionChecker implements PermissionChecker{
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: NIT: add space before the `{`?

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    mInodeTree = Preconditions.checkNotNull(inodeTree, "inodeTree");
+    mPermissionCheckEnabled =
+        Configuration.getBoolean(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_ENABLED);
+    mFileSystemSuperGroup =
+        Configuration.get(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP);
+    mInodePermissionChecker = InodePermissionChecker.create();
+  }
+
+  /**
+   * Checks whether a user has permission to perform a specific action on the parent of the given
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: NIT: does the interface already have these comments? If so, you can remove these.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java (hunk 1 above)
Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: Done.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java (hunk 2 above)
Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: Done.

**Comment by bf8086** [comment]
@gpang PTAL.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Default implementation of permission checker.
+ */
+// TODO(peis): Migrate this class to a set of static functions.
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: I think you can remove this TODO

Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: Done

**Pull request merged**